source/pod.go

```diff
@@ -154,7 +154,28 @@ func (ps *podSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, error
 		if ps.podSourceDomain != "" {
 			domain := pod.Name + "." + ps.podSourceDomain
 			if len(targets) == 0 {
-				addToEndpointMap(endpointMap, domain, suitableType(pod.Status.PodIP), pod.Status.PodIP)
+				ept := EndpointsTypePodIP
+				if tmp, ok := pod.Annotations[endpointsTypeAnnotationKey]; ok {
+					ept = tmp
+				}
+
+				switch ept {
+				case EndpointsTypeNodeExternalIP:
+					node, _ := ps.nodeInformer.Lister().Get(pod.Spec.NodeName)
+					for _, address := range node.Status.Addresses {
+						recordType := suitableType(address.Address)
+						// IPv6 addresses are labeled as NodeInternalIP despite being usable externally as well.
```

Comment:
This code overall adds to complexity of this function. ~100 lines all together now.

Can we extract logic, something like
```go
func processEndpointType(ps *podSource, pod *corev1.Pod, domain string, endpointMap map[endpoint.EndpointKey][]string) {
	ept := EndpointsTypePodIP
	if tmp, ok := pod.Annotations[endpointsTypeAnnotationKey]; ok {
		ept = tmp
	}

	switch ept {
	case EndpointsTypeNodeExternalIP:
		handleNodeExternalIP(ps, pod, domain, endpointMap)
	case EndpointsTypePodIP, EndpointsTypeHostIP:
		handlePodIP(pod, domain, endpointMap)
	default:
		log.Debugf("skipping pod %s. unknown endpoint type %s", pod.Name, ept)
	}
}
```